### ISSUE TITLE
Inconsistent font rendering

### ISSUE THREAD

**Issue by dimitry-ishenko** [microsoft/vscode-jupyter#8763]
Issue Type: <b>Bug</b>

1. Create new Jupyter notebook
2. Create Markdown cell
3. Enter in the cell:

Default Markdown text
## Heading 2
# Heading 1

4. Observe ginormous headings.

Extension version: 2022.1.1001614873
VS Code version: Code - Insiders 1.64.0-insider (23664355e61f3fb614932a5cb16613b4d987cfb7, 2022-01-21T13:26:52.572Z)
OS version: Linux x64 5.13.0-25-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz (8 x 1500)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|15.22GB (0.95GB free)|
|Process Argv|--unity-launch --crash-reporter-id 2273c5d1-5f55-44d1-afd8-13c81c1bf0ca|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details>

**Comment by dimitry-ishenko** [comment]
For comparison, this is how the Markdown preview extension renders the headings:

![Screenshot from 2022-01-17 18-40-34](https://user-images.githubusercontent.com/2858291/150572673-b26705f2-bd7f-452d-abe4-a2fb5992433f.png)

And here is how Google Colab renders them:

![Screenshot from 2022-01-17 18-52-10](https://user-images.githubusercontent.com/2858291/150572751-6e641cfd-4af9-4296-870e-c3101c17d6b1.png)

**Comment by dimitry-ishenko** [comment]
Notice that Heading 4 is nearly identical to the default text size. Whereas, Jupyter renders Heading 4 significantly larger.

Even further, notice how github renders Heading 4 below:

#### Heading 4

**Comment by dimitry-ishenko** [comment]
This is what I am seeing with Markdown preview for Heading 3 and Heading 4. The extension is keeping the default font sizes set by the browser.

![Screenshot from 2022-01-21 14-30-41](https://user-images.githubusercontent.com/2858291/150590044-82638420-ac4b-4f7c-a060-1ba949114237.png)

![Screenshot from 2022-01-21 14-30-49](https://user-images.githubusercontent.com/2858291/150590213-755d6dd1-7ab6-4593-8b11-7f7efbce0450.png)

**Comment by dimitry-ishenko** [comment]
Jupyter notebook on the other hand forces weird sizes. Why?

This seems to be the commit responsible for it: https://github.com/microsoft/vscode/commit/d1f2d0a7296a4747fe1ba4c3b01633e78392b15e

![Screenshot from 2022-01-21 14-33-24](https://user-images.githubusercontent.com/2858291/150590371-e80446bd-f407-4db2-b9ed-364672e37cc9.png)

![Screenshot from 2022-01-21 14-33-33](https://user-images.githubusercontent.com/2858291/150590398-8066fa00-5f5c-4900-b889-a890c6eaa32e.png)

**Comment by amunger** [comment]
From https://github.com/microsoft/vscode/issues/105900, it looks like they are tweaked mostly to keep consistent within the notebook markdown renderer.
Is the issue that it doesn't look the same as other renderers?

**Comment by dimitry-ishenko** [comment]
@amunger yes they look too big and inconsistent with (1) Markdown Preview extension, (2) Google Colab, (3) Jupyter, (4) GitHub, etc. See comparison in the beginning of the issue (1st and 2nd comments).

The complaint in https://github.com/microsoft/vscode/issues/105900 was that the headers were bold and had nothing to do with the size.

**Comment by dimitry-ishenko** [comment]
FWIW, haxxoring the following file fixes the sizes so that they match the rest of the world:

```diff
dimitry@linux:~$ diff -u /usr/share/code-insiders/resources/app/extensions/markdown-language-features/notebook-out/index.js{.orig,}
--- /usr/share/code-insiders/resources/app/extensions/markdown-language-features/notebook-out/index.js.orig	2022-01-21 19:13:23.050386852 -0500
+++ /usr/share/code-insiders/resources/app/extensions/markdown-language-features/notebook-out/index.js	2022-01-21 20:00:17.911510199 -0500
@@ -48,37 +48,7 @@
 			border-bottom: 2px solid;
 		}
 
-		h2, h3, h4, h5, h6 {
-			font-weight: normal;
-		}
-
-		h1 {
-			font-size: 2.3em;
-		}
-
-		h2 {
-			font-size: 2em;
-		}
-
-		h3 {
-			font-size: 1.7em;
-		}
-
-		h3 {
-			font-size: 1.5em;
-		}
-
-		h4 {
-			font-size: 1.3em;
-		}
-
-		h5 {
-			font-size: 1.2em;
-		}
-
-		h1,
-		h2,
-		h3 {
+		h1, h2, h3, h4, h5, h6 {
 			font-weight: normal;
 		}
```

**Comment by dimitry-ishenko** [comment]
Here is an actual patch to fix it in source:

```patch
From 70639788a864fac58a85674aeb979fb4ecb1e8fd Mon Sep 17 00:00:00 2001
From: Dimitry Ishenko <dimitry.ishenko@gmail.com>
Date: Fri, 21 Jan 2022 23:11:17 -0500
Subject: [PATCH] Fix Jupyter notebook header sizes

---
 .../notebook/index.ts                         | 32 +------------------
 1 file changed, 1 insertion(+), 31 deletions(-)

diff --git a/extensions/markdown-language-features/notebook/index.ts b/extensions/markdown-language-features/notebook/index.ts
index fe25756c0ea..57c88733b19 100644
--- a/extensions/markdown-language-features/notebook/index.ts
+++ b/extensions/markdown-language-features/notebook/index.ts
@@ -62,37 +62,7 @@ export const activate: ActivationFunction<void> = (ctx) => {
 			border-bottom: 2px solid;
 		}
 
-		h2, h3, h4, h5, h6 {
-			font-weight: normal;
-		}
-
-		h1 {
-			font-size: 2.3em;
-		}
-
-		h2 {
-			font-size: 2em;
-		}
-
-		h3 {
-			font-size: 1.7em;
-		}
-
-		h3 {
-			font-size: 1.5em;
-		}
-
-		h4 {
-			font-size: 1.3em;
-		}
-
-		h5 {
-			font-size: 1.2em;
-		}
-
-		h1,
-		h2,
-		h3 {
+		h1, h2, h3, h4, h5, h6 {
 			font-weight: normal;
 		}
 
-- 
2.32.0
```
